### ISSUE TITLE
Correct method of updating NodeJS

### ISSUE THREAD

**Issue by BKdilse** [MoneroOcean/node-cryptonight-hashing#18]
I've been running Node v6.9.2 for a long time, and now need to update to v8.11.3, so I can use this.

Has someone done this on a NodeJS-Pool setup?

I've tried:
```
nvm install v8.11.3
nvm alias default v8.11.3
```

This installs fine, but leaves the system unusable.  I'm guessing things like Path, PM2, etc.  is broken.

**Comment by bobbieltd** [comment]
1. npm update/install
2. pm2 restart xxxxx —update-env

**Comment by bobbieltd** [comment]
By the way, next time this issue should put at MoneroOcean’s nodejs-pool. Not this repo neither Snipa’s repo (Snipa does not maintain his repo anymore)

**Comment by BKdilse** [comment]
I've just upgraded Node, and pm2 was no longer working, so I re-installed it.
Rebooted, and so far Pool still works.  This time I removed the node_modules directory, and ran npm install (did not do this previously).

I then amended package.json, pointing to MO's hashing, and again deleted node_modules, and ran npm install.

Now i'm getting bad shares from miners, what would have changed in this module to cause this?

**Comment by bobbieltd** [comment]
You don’t need to remove old modules (only to free space but you will make you unable to undo to old node).
If I’m not wrong, Snipa’s repo uses MO’s utils, too.
Bad shares —> check CN multi-hashing in xmr.js —> return multiHashing.cryptonight(convertedBlob, 8);

**Comment by BKdilse** [comment]
Yeah, sticking to the version MO mentioned, v8.11.3.

coin config was working fine, until I changed to MO's hashing module.  Mine is using cnHashing.cryptonight(convertedBlob, 8)

**Comment by bobbieltd** [comment]
Is your CPU AMD ?

**Comment by BKdilse** [comment]
Nope, its Intel.  I'm currently using Venthos's hashing module, and it's working fine with that.

**Comment by bobbieltd** [comment]
If AMD CPU, try my repo https://github.com/bobbieltd/node-cryptonight-hashing

It seems MO mistyped variant for AMD CPU.

**Comment by bobbieltd** [comment]
So it is not the case

**Comment by bobbieltd** [comment]
Weird ! MO utils are used with my XMR fine.

**Comment by BKdilse** [comment]
Just tried this for Masari, and it's also showing bad shares for the Pool module.  Used with `return multiHashing.cryptonight(convertedBlob, 4);`

Any ideas, i'm now lost.
